### PR TITLE
Query Total: Add Border Support

### PR DESCRIPTION
**Pull request by shail-mehta**
## What?

Add Border support to the `Query Total` block.

Part of https://github.com/WordPress/gutenberg/issues/43247

## Why?
`Query Total` block is missing Border support.

## How?

Add the border support in block.json.

## Testing Instructions

- Go to Global Styles Settings. ( Under Appearance > Editor > Styles > Edit styles > Blocks )
- Make sure that `Query Total` block's border is Configurable via Global Styles.
- Edit Archives Template &  Add `Query Total` block and Apply the border Styles.
- Verify that `Query Total` block styles take precedence over global Styles.
- Verify that `Query Total` block borders display correctly in both the Editor and Frontend.

## Screenshots or Screencast 

https://github.com/user-attachments/assets/05a0b27e-38c1-4b3c-ac7e-ccf532c658cf

### CODE REVIEW

File: packages/block-library/src/query-total/block.json [hunk 1]
```diff
@@ -40,6 +40,13 @@
...
+		},
+		"__experimentalBorder": {
+			"radius": true,
+			"color": true,
+			"width": true,
+			"style": true
 		}
-	}
+	},
+	"style": "wp-block-query-total"
```
Comment (ramonjd) [inline packages/block-library/src/query-total/block.json:51]: Not sure this is required as [it's automatically set](https://github.com/WordPress/wordpress-develop/blob/27bb7ac2349eb81eb4e0ff5d275ff4ff3b98f3c4/src/wp-includes/blocks/index.php#L125-L127).

But there's no harm in leaving it.

**Review by ramonjd — APPROVED** [review]
Screenshot 2024-12-27 at 7 19 29 am


LGTM

Tested in global styles and at block level.

**Review by t-hamano — APPROVED** [review]
LGTM!

- Border styles defined in theme.json are applied to blocks.
- Border styles defined in theme.json are applied as initial values ​​in the global styles UI.
- Can change border styles via global styles.
- Can override border styles on block instances.

File: packages/block-library/src/query-total/block.json (hunk 1 above)
Comment (t-hamano) [reply packages/block-library/src/query-total/block.json:51]: I think this is fine for now. In the future, we may be able to remove the `style` and `editrStyle` fields from all core blocks.

**Comment by shail-mehta** [comment]
Thank you both for the review.

**Comment by aaronrobertshaw** [comment]
As noted over in https://github.com/WordPress/gutenberg/pull/68450#issuecomment-2572226411, the border support's default controls should match the spacing controls. As the padding/margin controls are displayed by default for this block, the border controls should be too.

@shail-mehta can you please spin up another PR to fix this?

**Comment by shail-mehta** [comment]
Created PR to Show Border Control Default:  [PR](https://github.com/WordPress/gutenberg/pull/68507)

**Pull request merged**